gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -115,12 +118,42 @@ private static JobSpec buildJobSpec(FlowSpec flowSpec, Config jobConfig, Long fl
       jobSpec.setConfig(jobSpec.getConfig().withValue(ConfigurationKeys.FLOW_FAILURE_OPTION,
           ConfigValueFactory.fromAnyRef(flowFailureOption)));
 
+      //Add tracking config to JobSpec.
+      addTrackingEventConfig(jobSpec, sysConfig);
+
       // Reset properties in Spec from Config
       jobSpec.setConfigAsProperties(ConfigUtils.configToProperties(jobSpec.getConfig()));
 
       return jobSpec;
     }
 
+    /**
+     * A method to add tracking event configurations to a JobSpec.
+     * This enables {@link org.apache.gobblin.metrics.GobblinTrackingEvent}s
+     * to be emitted from each Gobblin job orchestrated by Gobblin-as-a-Service, which will then be used for tracking the
+     * execution status of the job.
+     * @param jobSpec representing a fully resolved {@link JobSpec}.
+     */
+    private static void addTrackingEventConfig(JobSpec jobSpec, Config sysConfig) {
+      Config reportingConfig = ConfigUtils.getConfig(sysConfig, METRICS_REPORTING_CONFIG_PREFIX, ConfigFactory.empty());
+      if (!reportingConfig.isEmpty()) {
+        Config jobConfig = jobSpec.getConfig().withFallback(reportingConfig.atPath(METRICS_REPORTING_CONFIG_PREFIX));
```

Comment:
Why the fallback is required here? If user doesn't specify the metrics reporter, is it okay to reuse the gaas metrics reporter? I have concern because one is reporting from PROD network, but one is from Azkaban. They might have different settings.